docs/hello_nextflow/03_hello_plumbing.md

```diff
@@ -0,0 +1,755 @@
+
+### 1.2. Add a call to the new process in the workflow block
+
+Now we need to tell Nextflow to actually call the process that we just defined.
+
+In the workflow block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-plumbing.nf" linenums="53"
+    // emit a greeting
+    sayHello(greeting_ch)
+}
+```
+
+_After:_
+
+```groovy title="hello-plumbing.nf" linenums="53"
+    // emit a greeting
+    sayHello(greeting_ch)
+
+    // convert the greeting to uppercase
+    convertToUpper()
+}
+```
+
+This is not yet functional because we have not specified what should be input to the `convertToUpper()` process.
+
+### 1.3. Pass the output of the first process to the second process
+
+Now we need to connect the plumbing to make the output of the `sayHello()` process flow into the `convertToUpper()` process.
+
+Conveniently, Nextflow automatically packages the output of a process into a channel called `<process>.out`.
+So the output of the `sayHello` process is a channel called `sayHello.out`, which we can plug straight into the call to `convertToUpper()`.
+
+In the workflow block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-plumbing.nf" linenums="56"
+    // convert the greeting to uppercase
+    convertToUpper()
+}
+```
+
+_After:_
+
+```groovy title="hello-plumbing.nf" linenums="56"
+    // convert the greeting to uppercase
+    convertToUpper(sayHello.out)
+}
+```
+
+For a simple case like this (one output to one input), that's all we need to do to connect two processes!
+
+### 1.4. Run the workflow again with `-resume`
+
+Let's run this using the `-resume` flag, since we've already run the first step of the workflow successfully.
+
+```bash
+nextflow run hello-plumbing.nf -resume
+```
+
+There is now an extra line in the console output, which corresponds to the new process we just added:
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.10.0
+
+Launching `hello-plumbing.nf` [cheeky_hamilton] DSL2 - revision: f7fbe8e223
+
+executor >  local (3)
+[45/eb4757] sayHello (2)       [100%] 3 of 3, cached: 3 ✔
+[ae/4579ab] convertToUpper (3) [100%] 3 of 3 ✔
+```
+
+Have a look inside the work directory of one of the calls to the second process.
+
+```bash
+tree -a work/ae/4579ab5b4f2c1d986d3a955e31f2b7/
+```
+
+You should find two output files listed: the output of the first process, and the output of the second.
+
+```console title="Output"
+work/ae/4579ab5b4f2c1d986d3a955e31f2b7/
+├── Holà-output.txt -> /workspace/gitpod/hello-nextflow/work/dc/93eab52bd47ef198b1cfe1a7721b4b/Holà-output.txt
+└── UPPER-Holà-output.txt
+```
+
+The output of the first process is in there because Nextflow staged it there in order to have everything needed for execution within the same subdirectory.
+However, it is actually a symbolic link pointing to the the original file in the subdirectory of the first process call.
+By default, Nextflow uses symbolic links rather than copies to stage input and intermediate files.
```

Comment:
```suggestion
By default, when running on a single machine as we're doing here, Nextflow uses symbolic links rather than copies to stage input and intermediate files.
```

Gets more complex when we start talking about cloud etc.